client.c

```diff
@@ -1362,6 +1365,26 @@ ssize_t read(int fd, void *buf, size_t count) {
 		}
 
 		return count;
+	} else {
+		int t = get_timer_from_fd(fd);
+		if (t >= 0) {
+			if (count >= sizeof(timers[t].expired)) {
```

Comment:
This would look better if there was `else if ((t = get_timer_from_fd) >= 0)`.

Reply:
I'm not a fan of assign and compare. But I will modify for you.